docs/guide.ipynb

```diff
@@ -0,0 +1,180 @@
+    "\n",
+    "For example the following snippet shows how to use the `run_clingo` API to generate 1 recommendation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "lonely-cigarette",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'field': [{'name': 'temperature', 'type': 'number'},\n",
+      "           {'name': 'precipitation', 'type': 'number'}],\n",
+      " 'number_rows': 100,\n",
+      " 'task': 'value',\n",
+      " 'view': [{'mark': [{'channel': 'text',\n",
+      "                     'encoding': [{'channel': 'text',\n",
+      "                                   'field': ('f', 0),\n",
+      "                                   'scale_type': 'linear'},\n",
+      "                                  {'channel': 'size',\n",
+      "                                   'field': ('f', 1),\n",
+      "                                   'scale_type': 'linear'}],\n",
+      "                     'scale': ('s', 1),\n",
+      "                     'type': 'text'}],\n",
+      "           'scale': [{'channel': 'text', 'type': 'linear'},\n",
+      "                     {'channel': 'size', 'type': 'linear'}]}]}\n",
+      "[attribute(number_rows,root,100), attribute((field,name),(f,0),temperature), attribute((field,type),(f,0),number), attribute((field,name),(f,1),precipitation), attribute((field,type),(f,1),number), attribute((encoding,field),(e,0),(f,0)), attribute((encoding,field),(e,1),(f,1)), attribute(task,root,value), attribute((scale,type),(s,1),linear), attribute((scale,type),(s,0),linear), attribute((encoding,channel),(e,1),size), attribute((encoding,channel),(e,0),text), attribute((mark,type),(m,0),text), attribute((scale,channel),(s,1),size), attribute((scale,channel),(s,0),text), attribute((mark,channel),(m,0),size), attribute((mark,channel),(m,0),text), attribute((mark,scale),(m,0),(s,0)), attribute((mark,scale),(m,0),(s,1)), attribute((encoding,scale_type),(e,0),linear), attribute((encoding,scale_type),(e,1),linear), entity(field,root,(f,0)), entity(field,root,(f,1)), entity(view,root,(v,0)), entity(mark,(v,0),(m,0)), entity(encoding,(m,0),(e,0)), entity(encoding,(m,0),(e,1)), entity(scale,(v,0),(s,0)), entity(scale,(v,0),(s,1))]\n"
+     ]
+    }
+   ],
+   "source": [
+    "from draco import dict_to_facts, answer_set_to_dict, run_clingo\n",
+    "from draco.programs import define, hard, helpers, constraints, generate\n",
+    "from pprint import pprint\n",
+    "\n",
+    "prog = (\n",
+    "    generate.program\n",
+    "    + define.program\n",
+    "    + helpers.program\n",
+    "    + hard.program\n",
+    "    + constraints.program\n",
+    ")\n",
+    "with open(\"../draco/asp/examples/scatter.lp\") as file:\n",
```

Comment:
Let's define the program we want to demonstrate in the code instead of loading it from a file.